tests/test_diffraction_objects.py

```diff
@@ -341,5 +341,27 @@ def test_dump(tmp_path, mocker):
 
 @pytest.mark.parametrize("inputs, expected", tc_params)
 def test_constructor(inputs, expected):
-    actualdo = DiffractionObject(**inputs)
-    compare_dicts(actualdo.__dict__, expected)
+    actual_do = DiffractionObject(**inputs)
+    actual_dict = {
```

Comment:
we normally do this work using `actual_do.__dict__` so the test will fail if someone adds a new attribute to `DiffractionObject`

Reply:
yes, done.